gw_eccentricity/eccDefinition.py

```diff
@@ -295,19 +349,20 @@ def truncate_dataDict_if_necessary(self,
             truncation.
         """
         t = dataDict["t"]
-        phase22 = - np.unwrap(np.angle(dataDict["hlm"][(2, 2)]))
+        # Get phase22 depending on what is provided in dataDict
+        phase22, omega22, h22 = self.compute_phase22(dataDict)
```

Comment:
It's not nice that you have to do this here, and again after truncation.
Is there a better way? This is also a bit confusing.

Reply:
We discussed this earlier. The thing is that we need information about the full amp22 and phase22 to set the minimum width for peak_finding before truncating the dataDict. Once it passes the truncated dataDict, we use that to get waveform quantities to be used for extrema finding.